data/scenarios/Challenges/hackman.yaml

```diff
@@ -0,0 +1,231 @@
+version: 1
+name: Hackman
+author: Karl Ostmo
+description: |
+  Feed Hackman caffeine pellets
+creative: false
+attrs:
+  - name: ghost1
+    fg: "#ff0000"
+  - name: ghost2
+    fg: "#ffb8ff"
+  - name: ghost3
+    fg: "#00ffff"
+  - name: ghost4
+    fg: "#ffb852"
+objectives:
+  - teaser: Get pellets
+    goal:
+      - |
+        Pick up all of the caffeine "pellets" so that Hackman can write more code.
+      - |
+        Pay no mind to the colorful ghosts meandering about. That is, unless
+        you are feeling generous with your "donuts"...
+    condition: |
+      as base {
+        pcount <- count "pellet";
+        donut_count <- count "donut";
+        return $ pcount >= 181;
+      };
+  - id: donut_given
+    teaser: Give to ghost
+    hidden: true
+    optional: true
+    goal:
+      - |
+        Give a ghost a donut.
+    condition: |
+      def anyHasPowerup = \n.
+          let ghost_name = "ghost" ++ format n in
+          ghost <- robotnamed ghost_name;
+          hasPowerup <- as ghost {
+              has "donut";
+          };
+          
+          if hasPowerup {
+              return true;
+          } {
+              if (n > 1) {
+                  anyHasPowerup $ n - 1;
+              } {
+                  return false;
+              }
+          }
+          end;
+
+      anyHasPowerup 4;
+  - teaser: Magnanimity
+    hidden: true
+    optional: true
+    prerequisite: donut_given
+    goal:
+      - |
+        Give all ghosts a donut.
+    condition: |
+      def allHasPowerup = \n.
+          let ghost_name = "ghost" ++ format n in
+          ghost <- robotnamed ghost_name;
+          hasPowerup <- as ghost {
+              has "donut";
+          };
+          
+          if hasPowerup {
+              if (n > 1) {
+                  allHasPowerup $ n - 1;
+              } {
+                  return true;
+              }
+          } {
+              return false;
+          }
+          end;
+
+      allHasPowerup 4;
+robots:
+  - name: base
+    display:
+      orientationMap:
+        east: 'e'
```

Comment:
Is there a reason not to use `c` and `ↄ` for east and west?

Reply:
Agreed, these are probably better.